src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2641,16 +2762,213 @@ public override object PayloadValue(int index)
+            long size = (PointerSize == 8) ? GetInt64At(baseAddress + 8) : (long)GetInt32At(baseAddress + 8);
+            return new GCFitBucket(index, count, size);
+        }
+
+
+        #region Private
+        internal GCFitBucketInfoTraceData(Action<GCFitBucketInfoTraceData> action, int eventID, int task, string taskName, Guid taskGuid, int opcode, string opcodeName, Guid providerGuid, string providerName)
+            : base(eventID, task, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName)
+        {
+            Action = action;
+        }
+        protected internal override void Dispatch()
+        {
+            Action(this);
+        }
+        protected internal override Delegate Target
+        {
+            get { return Action; }
+            set { Action = (Action<GCFitBucketInfoTraceData>)value; }
+        }
+        protected internal override void Validate()
+        {
+            int size = 14 + Count * SizeOfGCFitBucket;
+            Debug.Assert(!(Version == 0 && EventDataLength != size));
+            Debug.Assert(!(Version > 0 && EventDataLength < size));
+        }
+        public override StringBuilder ToXml(StringBuilder sb)
+        {
+            Prefix(sb);
+            XmlAttrib(sb, "ClrInstanceID", ClrInstanceID);
+            XmlAttrib(sb, "BucketKind", BucketKind);
+            XmlAttrib(sb, "TotalSize", TotalSize);
+            XmlAttrib(sb, "Count", Count);
+            sb.AppendLine(">");
+            for (int i = 0; i < Count; i++)
+            {
+                Buckets(i).ToXml(sb).AppendLine();
+            }
+            sb.Append("</Event>");
+            return sb;
+        }
+
+        public override string[] PayloadNames
+        {
+            get
+            {
+                if (payloadNames == null)
+                {
+                    payloadNames = new string[] { "ClrInstanceID", "BucketKind", "TotalSize", "Count" };
+                }
+
+                return payloadNames;
+            }
+        }
+
+        public override object PayloadValue(int index)
+        {
+            switch (index)
+            {
+                case 0:
                     return ClrInstanceID;
+                case 1:
+                    return BucketKind;
+                case 2:
+                    return TotalSize;
+                case 3:
+                    return Count;
+
                 default:
                     Debug.Assert(false, "Bad field index");
                     return null;
             }
         }
 
-        private event Action<GenAwareEndTraceData> m_target;
+        // <struct name="Values"   count="Count"  >
+        //     <data name="Index" inType="win:UInt16" />
+        //     <data name="Count" inType="win:UInt32" />
+        //     <data name="Size" inType="win:Pointer" outType="win:HexInt64" />
+        // </struct>
+        private int SizeOfGCFitBucket
+        {
+            get
+            {
+                return PointerSize == 8 ? 16 : 12;
```

Comment:
My calculations are two bytes lower than what you have here.  Are you including two bytes for count?  From what I see here, count would not be included.

Reply:
The size is correct, the 2 bytes comes from alignment

Here is how the struct look like in 32 bits
```
0:000> dt coreclr!wks::etw_bucket_info
   +0x000 index            : Uint2B
   +0x004 count            : Uint4B
   +0x008 size             : Uint4B
```
and here is how the struct look like in 32 bits
```
0:000> dt coreclr!wks::etw_bucket_info
   +0x000 index            : Uint2B
   +0x004 count            : Uint4B
   +0x008 size             : Uint8B
```